### PR TITLE
Make LayoutLM tokenizers independent from BertTokenizer

### PR DESCRIPTION
**Pull request by arnaudstiegler**
# What does this PR do?

Decoupling `LayoutLMTokenizer` and `LayoutLMTokenizerFast` from `BertTokenizer`.

Since only a few class constants change between Bert and LayoutLM, there's a copy flag for every single method in the class.

I wonder whether having prefixes for the class constants could help reducing the amount of code, for instance:
`VOCAB_FILES_NAMES` -> `BERT_VOCAB_FILES_NAMES` 
so that we could simply use: `# Copied from transformers.models.bert.tokenization_bert_fast.BertTokenizerFast with Bert->LayoutLM` on the entire tokenizer class though I suspect there are good reasons not to do that.

Related to #19303 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by arnaudstiegler** [comment]
@sgugger The PR is ready for review :)

File: src/transformers/models/layoutlm/tokenization_layoutlm.py [hunk 1]
```diff
@@ -34,30 +38,491 @@
...
+def whitespace_tokenize(text):
+    """Runs basic whitespace cleaning and splitting on a piece of text."""
+    text = text.strip()
+    if not text:
+        return []
+    tokens = text.split()
+    return tokens
+
+
+class LayoutLMTokenizer(PreTrainedTokenizer):
```
Comment (sgugger) [inline src/transformers/models/layoutlm/tokenization_layoutlm.py:74]: I think you can apply the Copied from on the entire class without any issue, `VOCAB_FILES_NAMES` and all other constants will use the value defined in this module.

File: src/transformers/models/layoutlm/tokenization_layoutlm_fast.py [hunk 2]
```diff
@@ -43,31 +47,165 @@
...
 
-
 PRETRAINED_INIT_CONFIGURATION = {
     "microsoft/layoutlm-base-uncased": {"do_lower_case": True},
     "microsoft/layoutlm-large-uncased": {"do_lower_case": True},
 }
 
 
-class LayoutLMTokenizerFast(BertTokenizerFast):
+class LayoutLMTokenizerFast(PreTrainedTokenizerFast):
```
Comment (sgugger) [inline src/transformers/models/layoutlm/tokenization_layoutlm_fast.py:61]: Same here!

**Review by sgugger — COMMENTED** [review]
Thanks a lot for your PR! Very clean, but I think we can already apply the Copied from to the whole class. If not, would you mind explaining again what doesn't work? The constants shouldn't be any problem.

File: src/transformers/models/layoutlm/tokenization_layoutlm.py (hunk 1 above)
Comment (arnaudstiegler) [reply src/transformers/models/layoutlm/tokenization_layoutlm.py:75]: Oh yeah, didn't realize that! Added Copied statement to the class :)

File: src/transformers/models/layoutlm/tokenization_layoutlm_fast.py (hunk 2 above)
Comment (arnaudstiegler) [reply src/transformers/models/layoutlm/tokenization_layoutlm_fast.py:62]: Fixed!

**Review by sgugger — APPROVED** [review]
Perfect, thanks a lot!

**Pull request merged**